test/parallel_api/ranges/std_ranges_transform.pass.cpp

```diff
@@ -22,13 +22,40 @@ main()
-    test_range_algo<3, P2, data_in_out, false/*return type check*/>{}(dpl_ranges::transform, std::ranges::transform, f, &P2::proj);
+    //A checker below modifies a return type; a range based version with policy has another return type.
+    auto transform_unary_checker = [](std::ranges::random_access_range auto&& __r_in,
+                                      std::ranges::random_access_range auto&& __r_out, auto&&... args)
+    {
+        auto res = std::ranges::transform(std::forward<decltype(__r_in)>(__r_in), std::ranges::begin(__r_out),
+            std::forward<decltype(args)>(args)...);
 
-    test_range_algo<4, int, data_in_in_out, false/*return type check*/>{}(dpl_ranges::transform, std::ranges::transform, binary_f);
-    test_range_algo<5, int, data_in_in_out, false/*return type check*/>{}(dpl_ranges::transform, std::ranges::transform, binary_f, proj, proj);
+        using ret_type = std::ranges::unary_transform_result<std::ranges::borrowed_iterator_t<decltype(__r_in)>,
+            std::ranges::borrowed_iterator_t<decltype(__r_out)>>;
+        return ret_type{res.in, res.out};
+    };
+
+    test_range_algo<0, int, data_in_out>{}(dpl_ranges::transform, transform_unary_checker, f);
+    test_range_algo<1, int, data_in_out>{}(dpl_ranges::transform, transform_unary_checker, f, proj);
+    test_range_algo<2, P2, data_in_out>{}(dpl_ranges::transform, transform_unary_checker, f, &P2::x);
+    test_range_algo<3, P2, data_in_out>{}(dpl_ranges::transform, transform_unary_checker, f, &P2::proj);
+
+    //A checker below modifies a return type; a range based version with policy has another return type.
+    auto transform_binary_checker = [](std::ranges::random_access_range auto&& __r_1,
+                                       std::ranges::random_access_range auto&& __r_2,
+                                       std::ranges::random_access_range auto&& __r_out, auto&&... args)
+    {
+        auto res = std::ranges::transform(std::forward<decltype(__r_1)>(__r_1), std::forward<decltype(__r_2)>(__r_2),
+            std::ranges::begin(__r_out), std::forward<decltype(args)>(args)...);
+
+        using ret_type = std::ranges::binary_transform_result<std::ranges::borrowed_iterator_t<decltype(__r_1)>,
+            std::ranges::borrowed_iterator_t<decltype(__r_2)>, std::ranges::borrowed_iterator_t<decltype(__r_out)>>;
+        return ret_type{res.in1, res.in2, res.out};
+    };
+
+    test_range_algo<4, int, data_in_in_out>{}(dpl_ranges::transform, transform_binary_checker, binary_f);
+    test_range_algo<5, int, data_in_in_out>{}(dpl_ranges::transform, transform_binary_checker, binary_f, proj, proj);
+    test_range_algo<6, int, data_in_in_out>{}(dpl_ranges::transform, transform_binary_checker, binary_f, &P2::x, &P2::x);
+    test_range_algo<7, int, data_in_in_out>{}(dpl_ranges::transform, transform_binary_checker, binary_f, &P2::proj, &P2::proj);
```

Comment:
I guess there should be P2 type instead of int

Reply:
yes... it is a kind of local "merge/rebase" artefact....